sys/hacl/build.rs

```diff
@@ -38,55 +38,49 @@ fn includes(home_dir: &Path, include_str: &str) -> Vec<String> {
     ]
 }
 
-// FIXME: drop all the host specific checks and move them to target.
-#[cfg(all(
-    any(target_arch = "x86", target_arch = "x86_64"),
-    any(
-        target_os = "linux",
-        target_os = "macos",
-        all(target_os = "windows", any(target_env = "msvc", target_env = "gnu"))
-    )
-))]
-fn append_aesgcm_flags(flags: &mut Vec<String>) {
-    if cfg!(not(target_env = "msvc")) {
+#[allow(dead_code)]
+fn append_aesgcm_flags(platform: &Platform, flags: &mut Vec<String>) {
+    if platform.target_env != "msvc" {
         flags.push("-maes".to_string());
         flags.push("-mpclmul".to_string());
     }
 }
 
-fn append_simd128_flags(flags: &mut Vec<String>, is_bindgen: bool) {
-    if cfg!(any(target_arch = "x86", target_arch = "x86_64"))
-        && (is_bindgen || cfg!(not(target_env = "msvc")))
+#[allow(dead_code)]
+fn append_simd128_flags(platform: &Platform, flags: &mut Vec<String>, is_bindgen: bool) {
+    if platform.target_arch == "x86"
+        || platform.target_arch == "x86_64" && (is_bindgen || platform.target_env != "msvc")
```

Comment:
`&&` operator has precedence over `||` in Rust, I think we should set `platform.target_arch == "x86" || platform.target_arch == "x86_64"` in parentheses.